File: 228-summary-ranges/summary-ranges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> result;
        int left = 0;
        if (nums.size() == 0)
            return {};
            
        for (auto right = 1; right <= nums.size(); right++) {
            if (right == nums.size() || nums[right] != nums[right - 1] + 1) {
                if (left == right - 1) 
                    result.push_back(to_string(nums[left]));
                else 
                    result.push_back(to_string(nums[left]) + "->" + to_string(nums[right - 1]));
                left = right;
            }
        }
        return result;
    }
};

File: 228-summary-ranges/summary-ranges.go
func summaryRanges(nums []int) []string {
    l, r := 0, 0
    res := []string{}

    for l < len(nums) {
        for r < len(nums) - 1 && nums[r] + 1 == nums[r + 1]{
            r++
        }

        if l != r {
            res = append(res, strconv.Itoa(nums[l]) + "->" + strconv.Itoa(nums[r]))
        } else {
            res = append(res, strconv.Itoa(nums[l]))
        }

        r++
        l = r
    }

    return res 
}
